http/webserver.go

```diff
@@ -409,7 +414,40 @@ func HandleForward(h *config.WebForward, w http.ResponseWriter, r *http.Request,
 		proto = "https"
 	}
 	r.Header["X-Forwarded-Proto"] = []string{proto}
-	// todo: add Forwarded header? is anyone using it?
+	if isWebSocketRequest(r) {
+		// Create a new websocket server
+		websocketHandler := websocket.Handler(func(ws *websocket.Conn) {
```

Comment:
i think this is the right place to do this. but that we should monitor the websocket handshake ourselves (send it to the backend, read its response, check that it understood this was a websocket upgrade request, and send the response back the client). If the websocket upgrade was successful, we would then just shuffle the bytes between the two endpoints until the connection is closed. I don't think we have to enforce that the framing messages are correct.